.github/workflows/ci.yml

```diff
@@ -18,10 +18,10 @@ jobs:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       - name: Install poetry
         run: pipx install poetry
-      - name: Set up Python 3.8
+      - name: Set up Python
         uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
         with:
-          python-version: '3.8'
+          python-version: '3.9'
           cache: 'poetry'
       - name: Build
         run: poetry install --with dev,test
```

Comment:
I am not sure if this is actually using python 3.9. I have only learned recently that the ubuntu-latest GH image already have python installed and poetry will use that one as default and not 3.9.
I would advise adding a step before the `Build`: `poetry env info`.
To fix this, i think we need `poetry env use 3.9`